File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int heightSize){
    int i = 0;
    int j = heightSize-1;
    int water = 0;
    int minheight = 0;

    while(i<j){
        minheight = (height[i]<height[j]) ? height[i] : height[j];
        water = (water<(minheight)*(j-i)) ? (minheight)*(j-i) : water;
        if (height[i]<height[j]){
            i++;
        } else {
            j--;
        }
    }
    return water;
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # two pointer로 양쪽에서 오면서 짧은 기둥 기준으로 곱하기 -> 더 많으면 max area 덮어쓰기
        l = 0
        r = len(height)-1
        maxarea = 0
        
        while l<r:
#            maxarea = max(maxarea,min(height[r],height[l])*(r-l) )
            if maxarea < min(height[r],height[l])*(r-l):
                maxarea = min(height[r],height[l])*(r-l)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return maxarea